presto-main/src/main/java/com/facebook/presto/Session.java

```diff
@@ -119,7 +121,8 @@ public Session(
             Map<String, String> preparedStatements,
             Map<SqlFunctionId, SqlInvokedFunction> sessionFunctions,
             Optional<Tracer> tracer,
-            WarningCollector warningCollector)
+            WarningCollector warningCollector,
+            CanonicalPlanHashes canonicalPlanHashes)
```

Comment:
why are the plan hashes part of a session?  It doesn't seem right. A session can be shared across multiple queries.

Reply:
I didn't realize a session can be shared, going to store it in plan node

Reply:
actually, sorry, this was my bad. The session concept is shared across queries (e.g. client might send the same session info for all queries it submits), but the session object itself is per-query